src/Microsoft.AspNet.Hosting/WebHostOptions.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.AspNet.Hosting
+{
+    public class WebHostOptions
+    {
+        private const string ApplicationKey = "app";
+        private const string DetailedErrorsKey = "detailederrors";
+        private const string EnvironmentKey = "environment";
+        private const string OldEnvironmentKey = "ENV";
+        private const string ServerKey = "server";
+        private const string WebRootKey = "webroot";
+
+        public WebHostOptions()
+        {
+        }
+
+        public WebHostOptions(IConfiguration configuration)
+        {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            Application = configuration[ApplicationKey];
+            DetailedErrors = string.Equals("true", configuration[DetailedErrorsKey], StringComparison.OrdinalIgnoreCase)
+                || string.Equals("1", configuration[DetailedErrorsKey], StringComparison.OrdinalIgnoreCase);
+            Environment = configuration[EnvironmentKey] ?? configuration[OldEnvironmentKey];
+            Server = configuration[ServerKey];
+            WebRoot = configuration[WebRootKey];
```

Comment:
So if you wanted to use the config binder, it would look something more like this, the only weirdness is with the Environment/DetailedErrors properties which need a method to add the logic.  

```
public class WebHostOptions {
    public string App { get; set; }
    public string Server {get; set; }
   public string WebRoot {get; set; }
   public string DetailedErrors {get; set; }
   public string Environment {get; set; }
   public string Env {get; set; }

   public bool ShowDetailedErrors { get {
      return string.Equals("true", DetailedErrors)
                || string.Equals("1", DetailedErrors) ;
   } }

   public string ResolveEnvironment() {
      return Environment ?? Env;
   }
}

To get it out of config assuming its at the root level:
   config.Get<WebHostOptions>();
```